src/steamship/agents/schema/chathistory.py

```diff
@@ -1,30 +1,48 @@
 from __future__ import annotations
 
-from typing import Dict, List, Optional, Union
+import uuid
+from typing import Dict, List, Optional, Union, cast
 
-from steamship import File, MimeTypes, SteamshipError
+from steamship import File, MimeTypes, Steamship, SteamshipError, Task
 from steamship.agents.schema.message_selectors import MessageSelector
+from steamship.agents.schema.text_splitters import FixedSizeTextSplitter, TextSplitter
 from steamship.base.client import Client
 from steamship.data import TagKind
 from steamship.data.block import Block
+from steamship.data.plugin.index_plugin_instance import EmbeddingIndexPluginInstance, SearchResults
 from steamship.data.tags import Tag
 from steamship.data.tags.tag_constants import ChatTag, DocTag, RoleTag, TagValueKey
 
 
 class ChatHistory:
-    """A ChatHistory is a wrapper of a File ideal for ongoing interactions between a user and a virtual assistant."""
+    """A ChatHistory is a wrapper of a File ideal for ongoing interactions between a user and a virtual assistant.
+    It also includes vector-backed storage for similarity-based retrieval."""
 
     file: File
+    embedding_index: EmbeddingIndexPluginInstance
+    text_splitter: TextSplitter
 
-    def __init__(self, file: File):
+    def __init__(
+        self,
+        file: File,
+        embedding_index: Optional[EmbeddingIndexPluginInstance],
+        text_splitter: TextSplitter = None,
```

Comment:
@dkolas `text_splitter` stuff looks good to me.

i'll be curious to see what we find out about chat history messages (typical sizes, etc.). _Might_ be worth `logging.info` with an `extra` field related to number of chunks for new messages or something that would allow us to have some insights (as a platform) and to provide those insights to our customers. Unless we think DB tag queries should suffice moving forward as an alternative (for platform usages)?